packages/x-components/src/x-modules/facets-next/entities/single-select.modifier.ts

```diff
@@ -0,0 +1,48 @@
+import { isFacetFilter, FacetFilter, Facet } from '@empathyco/x-types-next';
+import { FacetsNextGetters } from '../store/types';
+import { BaseFilterEntityModifier } from './types';
+
+/**
+ * Allows only to select only one filter from the same facet at the same time.
+ */
+export class SingleSelectModifier extends BaseFilterEntityModifier {
+  /**
+   * Selects the passed filter, and then deselects any other filter of the same facet.
+   *
+   * @param filter - The filter to select.
+   */
+  select(filter: FacetFilter): void {
+    this.getOtherFilters(filter).forEach(siblingFilter => this.deselect(siblingFilter));
+    this.entity.select(filter);
+  }
+
+  /**
+   * Retrieves the rest of selected filters of the same facet than the one given.
+   *
+   * @param filter - The filter to find its relatives.
+   * @returns A list of selected filters that belong to the same facet than the filter passed.
+   * @internal
+   */
+  protected getOtherFilters(filter: FacetFilter): FacetFilter[] {
+    return isFacetFilter(filter)
```

Comment:
This seems dumb because we are checking if a `FacetFilter` is a `FacetFilter`. The problem is that when you decorate some entity using the factory you loos this `FacetFilter` type in the select method. This decorator shouldn't be used with `RawFilters`, but we added this simple check as a crash prevention measure.